### PR TITLE
[BEAM-6908] Support Python3 performance benchmarks - part 2

### PR DESCRIPTION
**Pull request by markflyhigh**
Support Python3 performance benchmarks - part 2

Keep use of Perfkit performance infrastructure for Python3 benchmarking due to exist integration with Python pipeline and automated metrics publishing. Part 1 changes are on PerfkitBenchmarker in https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1900/commits/d0ad984e2ae4d836fc7d6440e44104d2b469ec98

Changes include in this PR:
- Move python performance test parameters to new class `PythonPerformanceTestConfiguration` to make task construction clear and easier to maintain.
- Create `enablePythonPerformanceTest` in `BeamModulePlugin` so that people can enable python performance test in different projects by simply calling this function.
- Enable performance tests in Python3 project `sdks/python/test-suites/dataflow/py3`
- Fix a bug in `run_integration_test.sh` due to wrong check order.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by markflyhigh** [comment]
+R: @tvalentyn

**Comment by udim** [comment]
run python postcommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -1722,6 +1740,45 @@ class BeamModulePlugin implements Plugin<Project> {
...
+
+        project.task('integrationTest') {
+          dependsOn 'installGcpTest'
+          dependsOn 'sdist'
+
+          doLast {
+            def argMap = [:]
+
+            // Build test options that configures test environment and framework
+            def testOptions = []
```
[the remark this answers is not in the document]
Comment (markflyhigh) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Currently `basicTestOpts` is defined under subproject build.gradle and the major purpose is to set `"--processes"` and `"--process-timeout"` which is used to parallel multiple integration test run.

In performance testing, we mainly run single integration test for benchmarking, so I don't see much use case here. On the other hand, people can use `extraTestOptions` if needed.

**Comment by markflyhigh** [comment]
Thank you. postcommit run failed consistently for a while and got fixed recently (today?). I'll rebase and rerun them.

**Comment by markflyhigh** [comment]
run python postcommit

**Comment by markflyhigh** [comment]
Perfkit corresponding changes are merged in https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/pull/1900/commits/d0ad984e2ae4d836fc7d6440e44104d2b469ec98. We now can merge this PR after tests pass.

**Comment by markflyhigh** [comment]
Run Java PreCommit

**Comment by markflyhigh** [comment]
Run Java PreCommit

**Comment by markflyhigh** [comment]
Rebased this PR against master.

**Comment by markflyhigh** [comment]
Run Python_PVR_Flink PreCommit

**Pull request merged**